packages/contracts/test/voting/multisig.ts

```diff
@@ -338,140 +437,189 @@ describe('Multisig', function () {
       ).to.equal(id);
     });
 
-    it('reverts when approving multiple times', async () => {
-      await multisig.approve(id, true);
+    describe('canApprove:', async () => {
+      it('returns `false` if the proposal is already executed', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        await multisig.connect(signers[1]).approve(id, false);
+        await multisig.connect(signers[2]).approve(id, true);
 
-      // Try to vote again
-      await expect(multisig.approve(id, true)).to.be.revertedWith(
-        customError('ApprovalCastForbidden', id, signers[0].address)
-      );
-    });
+        expect((await multisig.getProposal(id)).executed).to.be.true;
 
-    it('reverts if minimal approval is not met yet', async () => {
-      await expect(multisig.execute(id)).to.be.revertedWith(
-        customError('ProposalExecutionForbidden', id)
-      );
-    });
+        expect(await multisig.canApprove(id, signers[3].address)).to.be.false;
+      });
 
-    it('executes if the minimum approval is met', async () => {
-      await multisig.connect(signers[0]).approve(id, false);
-      await multisig.connect(signers[1]).approve(id, false);
-      await multisig.connect(signers[2]).approve(id, false);
+      it('returns `false` if the approver is not listed', async () => {
+        expect(await multisig.isListed(signers[9].address)).to.be.false;
 
-      const proposal = await multisig.getProposal(id);
+        expect(await multisig.canApprove(id, signers[9].address)).to.be.false;
+      });
 
-      expect(proposal.parameters.minApprovals).to.equal(minApprovals);
-      expect(await multisig.approvals(id)).to.be.eq(minApprovals);
+      it('returns `false` if the approver has already approved', async () => {
+        await multisig.connect(signers[0]).approve(id, false);
+        expect(await multisig.canApprove(id, signers[0].address)).to.be.false;
+      });
 
-      expect(await multisig.canExecute(id)).to.equal(true);
-      await expect(multisig.execute(id)).to.not.be.reverted;
+      it('returns `false` if the approver is listed', async () => {
```

Comment:
wrong test name ? it returns true.

Reply:
Indeed, fixed in commit 560908ca218e50cbe90f8769e05f8c17e33d6096